src/v/storage/disk_log_impl.cc

```diff
@@ -1141,7 +1141,7 @@ ss::future<bool> disk_log_impl::update_start_offset(model::offset o) {
     // never be sure that we are not overwriting a bigger value that
     // is concurrently written to kvstore.
     return _start_offset_lock.with([this, o] {
-        if (o <= _start_offset) {
+        if (o < _start_offset) {
```

Comment:
I feel that this patch is a bit counter-intuitive. Maybe it will be more clear and robust to simply check in `disk_log_impl::do_prefix_truncate` that we still have segments to delete even if start_offset doesn't need to change?

Reply:
i agree, i will change that accordingly